### ISSUE TITLE
packing with pyinstaller problems

### ISSUE THREAD

**Issue by vin7ent** [joke2k/faker#891]
When I use pyinstaller to  pack the file to exe, it turns out the problem:
ImportError: No module named providers.

Then I added the hidden imports parameters, errors changes to
Invalid configuration for faker locale 'en_US'

So how to fix it?

**Comment by arossert** [comment]
I have the same issue, there are some debaites regarding this on the PyInstaller github but I have not been able to make it work yet.
You can follow it here [#3465](https://github.com/pyinstaller/pyinstaller/pull/3465) and here [#3839](https://github.com/pyinstaller/pyinstaller/issues/3839).

It seems that there is a code in faker that check if the code is "frozen" or not.

Any help with making this work will be higly appriciated.

**Comment by fcurella** [comment]
@arossert I've created a branch at https://github.com/joke2k/faker/tree/hardcode-locales that removes the dynamic loading of some values. There's still some dynamic loading happening, but hopefully hardcoding the values in `config` is enough to work. Could you give it a try?

**Comment by arossert** [comment]
@fcurella Thanks, I have tested your branch and it is working with this hook:
```
from PyInstaller.utils.hooks import collect_submodules, collect_data_files

hiddenimports = collect_submodules('faker.providers')
datas = collect_data_files('text_unidecode')
```
I'm not sure that the creator will want to change this values to be hardcoded, if so it will make the like much easier.

Following this approch #704, we might need some code to detect the useage of PyInstaller (maybe check for `sys._MEIPASS` attribute) and look for the files relative to it's path, then in the PyInstaller hook we can include the src files.

**Comment by arossert** [comment]
@joke2k Any update regarding this? I really need to pack faker with my project using PyInstaller and don't want to use a branch.

**Comment by fcurella** [comment]
@arossert there's a problem with my PR.

As it's written, it would break some existing feature, specifically the command-line option to include custom providers.

The PR is a good starting point, but it needs some more work, an my time is limited. Any help would be appreciated :)

**Comment by arossert** [comment]
@fcurella thanks, I would love to help with this but don't know how to make it work.
I have tried to create a hook in PyInstaller for that and got some progress until I realized that `pkgutil.iter_modules()` is not supported in the frozen version.
Is there another way that you know to load the package list dynamically? In my hook, all the providers code is available so I can iterate the directory tree.

**Comment by arossert** [comment]
@fcurella I really need this to work, is my changes seem reasonable?

**Comment by fcurella** [comment]
@arossert Could you submit a PR and we'll discuss it from there?

**Comment by fcurella** [comment]
closed in #927
